datadog/fwprovider/resource_datadog_app_builder_app.go

```diff
@@ -0,0 +1,720 @@
+	if err != nil {
+		err = fmt.Errorf("error replacing connection IDs in queries: %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func overrideAppBuilderAppAttributesInUpdateRequestAttributes(plan appBuilderAppResourceModel, appJsonMap map[string]interface{}) error {
+	// name, description, root_instance_name are straightforward string replacements
+	if !plan.Name.IsNull() && !plan.Name.IsUnknown() {
+		appJsonMap["name"] = plan.Name.ValueString()
+	}
+	if !plan.Description.IsNull() && !plan.Description.IsUnknown() {
+		appJsonMap["description"] = plan.Description.ValueString()
+	}
+	if !plan.RootInstanceName.IsNull() && !plan.RootInstanceName.IsUnknown() {
+		appJsonMap["rootInstanceName"] = plan.RootInstanceName.ValueString()
+	}
+
+	// Using override_action_query_names_to_connection_ids, replace connection ids in the update request attributes with the ones provided in the plan
+	err := replaceConnectionIDsInActionQueries(plan.OverrideActionQueryNamesToConnectionIDs, appJsonMap)
+	if err != nil {
+		err = fmt.Errorf("error replacing connection IDs in queries: %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// replace the connection ids in the queries with the ones provided in the plan, as specified by {action_query_name: connection_id}
+func replaceConnectionIDsInActionQueries(overrideActionQueryNamesToConnectionIDs types.Map, appJsonMap map[string]interface{}) error {
+	// skip if overrideActionQueryNamesToConnectionIDs is empty
+	if overrideActionQueryNamesToConnectionIDs.IsNull() || overrideActionQueryNamesToConnectionIDs.IsUnknown() {
+		return nil
+	}
+
+	// Handle connection ID overrides
+	queries, ok := appJsonMap["queries"].([]interface{})
+	if !ok {
+		return nil
+	}
+
+	// Create a properly typed map of overrides
+	overrides := make(map[string]string)
+	for k, v := range overrideActionQueryNamesToConnectionIDs.Elements() {
+		strVal, ok := v.(types.String)
+		if !ok {
+			return fmt.Errorf("expected string value for key %s, got %T", k, v)
+		}
+		overrides[k] = strVal.ValueString()
+	}
+
+	// Track which overrides have been used
+	unusedOverrides := maps.Clone(overrides)
+
+	// Update connection IDs in queries
+	for _, q := range queries {
+		query, ok := q.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		queryName, ok := query["name"].(string)
+		if !ok {
+			continue
+		}
+
+		// Only process action queries
+		if queryType, ok := query["type"].(string); !ok || queryType != "action" {
+			continue
+		}
+
+		// Check if we have an override for this query
+		if connectionID, ok := overrides[queryName]; ok {
+			properties, ok := query["properties"].(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			spec, ok := properties["spec"].(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			spec["connectionId"] = connectionID
+			delete(unusedOverrides, queryName)
+		}
+	}
+
+	// Return error if any overrides weren't used
```

Comment:
suggestion: this could just be a no-op instead of an error

Reply:
hm, I thought this would be helpful for users who may have accidentally misspelled a query name and want to quickly figure out why a connectionID wasn't changed as expected

Reply:
sounds reasonable to me